From a9e102abcb14668d1060e1aa1106baf7b46f6e9a Mon Sep 17 00:00:00 2001
From: zhuw <zhuw@codeaurora.org>
Date: Tue, 19 Dec 2017 15:53:26 +0800
Subject: [PATCH 21/67] SnapdragonGallery: Fix multithread synchronization in
 screenReceiver

add judgement condition when receive intent about screen off and on.

Change-Id: I32b288ed29261999284df5d6d6857cf3573bc5e6
CRs-Fixed: 2158799
---
 src/com/android/gallery3d/app/MovieActivity.java | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/com/android/gallery3d/app/MovieActivity.java b/src/com/android/gallery3d/app/MovieActivity.java
index f5e7a65fb..1e9fa999a 100644
--- a/src/com/android/gallery3d/app/MovieActivity.java
+++ b/src/com/android/gallery3d/app/MovieActivity.java
@@ -751,12 +751,12 @@ public class MovieActivity extends AbstractPermissionActivity {
             }
             if (Intent.ACTION_SCREEN_OFF.equals(intent.getAction())) {
                 // Only stop video.
-                if (mControlResumed) {
+                if (mControlResumed && !mResumed) {
                     mPlayer.onStop();
                     mControlResumed = false;
                 }
             } else if (Intent.ACTION_USER_PRESENT.equals(intent.getAction())) {
-                if (!mControlResumed) {
+                if (!mControlResumed && mResumed) {
                     mPlayer.onResume();
                     mControlResumed = true;
                 }
-- 
2.17.1

